clients/src/main/java/org/apache/kafka/common/network/KafkaChannel.java

```diff
@@ -369,6 +369,16 @@ public InetAddress socketAddress() {
         return transportLayer.socketChannel().socket().getInetAddress();
     }
 
+    /**
+     * Returns the port to which this channel's socket is connected or `null` if the socket has never been connected.
```

Comment:
nit: This cannot return `null` if the socket has never been connected.

Reply:
Thanks for pointing out, it should be 0 instead of null. Done.